engine/graphics/src/vulkan/graphics_vulkan.cpp

```diff
@@ -618,24 +618,31 @@ namespace dmGraphics
         TextureCreationParams default_texture_creation_params;
         default_texture_creation_params.m_Width          = 1;
         default_texture_creation_params.m_Height         = 1;
+        default_texture_creation_params.m_Depth          = 1;
         default_texture_creation_params.m_OriginalWidth  = default_texture_creation_params.m_Width;
         default_texture_creation_params.m_OriginalHeight = default_texture_creation_params.m_Height;
 
-        const uint8_t default_texture_data[] = { 255, 0, 255, 255 };
+        const uint8_t default_texture_data[4 * 6] = {0}; // RGBA * 6 (for cubemap)
 
         TextureParams default_texture_params;
         default_texture_params.m_Width  = 1;
         default_texture_params.m_Height = 1;
+        default_texture_params.m_Depth  = 1;
         default_texture_params.m_Data   = default_texture_data;
         default_texture_params.m_Format = TEXTURE_FORMAT_RGBA;
 
-        context->m_DefaultTexture = VulkanNewTexture(context, default_texture_creation_params);
-        VulkanSetTexture(context->m_DefaultTexture, default_texture_params);
+        context->m_DefaultTexture2D = VulkanNewTexture(context, default_texture_creation_params);
+        VulkanSetTexture(context->m_DefaultTexture2D, default_texture_params);
 
-        for (int i = 0; i < DM_MAX_TEXTURE_UNITS; ++i)
-        {
-            context->m_TextureUnits[i] = context->m_DefaultTexture;
-        }
+        default_texture_creation_params.m_Type  = TEXTURE_TYPE_2D_ARRAY;
+        default_texture_creation_params.m_Depth = 1;
+        context->m_DefaultTexture2DArray        = VulkanNewTexture(context, default_texture_creation_params);
+
+        default_texture_creation_params.m_Type  = TEXTURE_TYPE_CUBE_MAP;
+        default_texture_creation_params.m_Depth = 6;
+        context->m_DefaultTextureCubeMap        = VulkanNewTexture(context, default_texture_creation_params);
```

Comment:
We need to create default textures so that we can bind correct default types when a sampler has not been bound. OpenGL does this automatically but we need to be more explicit with vulkan.